pkg/controllers/k8sdefaultendpoint/k8sdefaultendpoint.go

```diff
@@ -74,51 +80,155 @@ func syncKubernetesServiceEndpoints(ctx context.Context, virtualClient client.Cl
 		return err
 	}
 
-	// get virtual service endpoints
-	vObj := &corev1.Endpoints{}
-	err = virtualClient.Get(ctx, types.NamespacedName{
-		Namespace: "default",
-		Name:      "kubernetes",
-	}, vObj)
-	if err != nil {
-		if kerrors.IsNotFound(err) {
+	vEndpoints := &corev1.Endpoints{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "default",
+			Name:      "kubernetes",
+		},
+	}
+
+	result, err := controllerutil.CreateOrPatch(ctx, virtualClient, vEndpoints, func() error {
+		if vEndpoints.Labels == nil {
+			vEndpoints.Labels = map[string]string{}
+		}
+		vEndpoints.Labels[discovery.LabelSkipMirror] = "true"
```

Comment:
Why do we need that?

Reply:
That's what k8s does - https://github.com/kubernetes/kubernetes/blob/7380fc735aca591325ae1fabf8dab194b40367de/pkg/controlplane/reconcilers/endpointsadapter.go#L206